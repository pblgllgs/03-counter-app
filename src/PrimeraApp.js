
const PrimeraApp = () => {
  return (
    <>
      <h1> Hola mundo </h1>
      <p>Mi primera app React</p>
    </>
  );
};

export default PrimeraApp;
