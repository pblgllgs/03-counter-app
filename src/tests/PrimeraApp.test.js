import PrimeraApp from "../PrimeraApp";
import React from "react";
import { shallow } from "enzyme";

describe("PrimeraApp test", () => {
  /* test('Deberia mostrar "Hola soy pablo"', () => {
    const msg = "Hola soy pablo";
    const { getByText } = render(<PrimeraApp saludo="Hola soy pablo" />);
    expect(getByText(msg)).toBeInTheDocument();
    //utilizando esa importaci√≥n
    import '@testing-library/jest-dom';
  }); */

  test("Debe mostrar <PrimeraApp /> correctamente", () => {
    const saludo = "Hola";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar el subtitulo mostrado por props", () => {
    const saludo = "Hola";
    const subtitulo = "Soy un subtitulo";
    const wrapper = shallow(
      <PrimeraApp 
        saludo={saludo}
        subtitulo={subtitulo}
      />);
    const textoParrafo = wrapper.find('p').text();
    expect(textoParrafo).toBe(subtitulo);
  });
});
