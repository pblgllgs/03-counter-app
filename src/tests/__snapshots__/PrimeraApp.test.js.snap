// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeraApp test Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola
    !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
